### PR TITLE
fix adding empty sequence to HTTP headers causes some problems

### PR DESCRIPTION
**Pull request by ringabout**
Allowing adding empty sequence causes some problems when visiting it. 
```nim
import httpcore

var headers = newHttpHeaders()
headers["empty"] = @[]


if headers.hasKey("empty"):
  # Error: unhandled exception: index out of bounds, the container is empty [IndexError]
  echo headers["empty"]
```

### CODE REVIEW

File: lib/pure/httpcore.nim [hunk 1]
```diff
@@ -166,9 +166,10 @@ proc `[]=`*(headers: HttpHeaders, key, value: string) =
 
 proc `[]=`*(headers: HttpHeaders, key: string, value: seq[string]) =
   ## Sets the header entries associated with ``key`` to the specified list of
-  ## values.
-  ## Replaces any existing values.
-  headers.table[headers.toCaseInsensitive(key)] = value
+  ## values. Replaces any existing values.
+  ## If ``value`` is an empty sequence, the key will be ignored.
+  if value.len > 0:
+    headers.table[headers.toCaseInsensitive(key)] = value
```
Comment (dom96) [inline lib/pure/httpcore.nim]: IMO you should remove the key instead.

**Review by dom96 — CHANGES_REQUESTED** [review]

File: lib/pure/httpcore.nim (hunk 1 above)
Comment (ringabout) [reply lib/pure/httpcore.nim]: yea, I agree.

Comment (juancarlospaco) [reply lib/pure/httpcore.nim]: @xflywind 
Check that some fields can be empty according to the standard, content length can be empty I think, is optional, I dont remember if theres others too.

Comment (ringabout) [reply lib/pure/httpcore.nim]: yea, it can be empty, but should use `headers.add(@[""])` instead.

Comment (juancarlospaco) [reply lib/pure/httpcore.nim]: :+1:

**Pull request closed without merging**